### PR TITLE
[WIP] New csproj

### PR DESCRIPTION
**Pull request by danielmarbach**
Connects to https://github.com/Particular/NServiceBus.Azure/issues/302

### CODE REVIEW

**Comment by danielmarbach** [comment]
What I see locally is the VS constantly tries to change the Solution file

From

```
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Debug|x64.ActiveCfg = Debug|x64
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Debug|x64.Build.0 = Debug|x64
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Release|x64.ActiveCfg = Release|x64
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Release|x64.Build.0 = Release|x64
```

to
```
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Debug|x64.ActiveCfg = Debug|Any CPU
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Debug|x64.Build.0 = Debug|Any CPU
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Release|x64.ActiveCfg = Release|Any CPU
		{6447FA5E-9C6B-4DFB-87BE-7399D6B4DFBF}.Release|x64.Build.0 = Release|Any CPU
```

@bording any ideas how to resolve this?

I tried to set RuntimeIdentifier, didn't help

File: src/Tests/app.config [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="nunit.framework" publicKeyToken="2638cd05610744eb" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.6.1.0" newVersion="3.6.1.0" />
```
Comment (SimonCropp) [inline src/Tests/app.config]: why is this required? and also the newVersion is wrong

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj [hunk 2]
```diff
@@ -1,107 +1,27 @@
...
-    <None Include="packages.config" />
-  </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+    <PackageReference Include="Microsoft.ServiceFabric" Version="5.6.210" />
+    <PackageReference Include="Microsoft.ServiceFabric.Data" Version="2.6.210"/>
+    <PackageReference Include="Microsoft.ServiceFabric.FabricTransport.Internal" Version="2.6.210"/>
+    <PackageReference Include="Microsoft.ServiceFabric.Services" Version="2.6.210"/>    
+    <PackageReference Include="Microsoft.ServiceFabric.Services.Remoting" Version="2.6.210"/>        
+    <PackageReference Include="NUnit" Version="3.7.1"/>            
+    <PackageReference Include="NUnitLite" Version="3.7.1"/>
```
Comment (SimonCropp) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: why NUnitLite?

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj [hunk 3]
```diff
@@ -1,107 +1,27 @@
...
-    <DefineConstants>TRACE</DefineConstants>
-    <Optimize>true</Optimize>
-    <DebugType>pdbonly</DebugType>
-    <PlatformTarget>x64</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
+    <DebugSymbols>True</DebugSymbols>
+    <WarningLevel>4</WarningLevel>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <DocumentationFile>$(ProjectDir)bin\$(Configuration)\$(AssemblyName).xml</DocumentationFile>
```
Comment (SimonCropp) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: so @bording pointed out a better way of doing the doco and build output https://github.com/Particular/NServiceBus.Spring/pull/30/commits/02e2ba75bfc02f4602440b37f85c4e51b22f2fd8

File: src/Tests/app.config (hunk 1 above)
Comment (danielmarbach) [inline src/Tests/app.config]: This is not required and will be removed

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj [hunk 4]
```diff
@@ -1,107 +1,27 @@
...
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">x64</Platform>
-    <ProjectGuid>{2A4EE72F-E17F-4858-ADAA-34FB47068F5C}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>NServiceBus.Persistence.TestRunner</RootNamespace>
+    <PlatformTarget>x64</PlatformTarget>    
+    <TargetFramework>net452</TargetFramework>
     <AssemblyName>NServiceBus.Persistence.TestRunner</AssemblyName>
```
Comment (SimonCropp) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: AssemblyName is by default inferred from the project file name. so this is redundant

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj (hunk 2 above)
Comment (danielmarbach) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: Test runner details because we are running tests inside the cluster. This functionality is only available in NUnit Lite

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj [hunk 5]
```diff
@@ -1,107 +1,27 @@
...
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
-    <OutputPath>bin\x64\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <Optimize>true</Optimize>
-    <DebugType>pdbonly</DebugType>
-    <PlatformTarget>x64</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
+    <DebugSymbols>True</DebugSymbols>
+    <WarningLevel>4</WarningLevel>
```
Comment (SimonCropp) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: this is the default. so can be removed

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj (hunk 4 above)
Comment (danielmarbach) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: Will address

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj (hunk 3 above)
Comment (danielmarbach) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: Will address

File: src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj (hunk 5 above)
Comment (danielmarbach) [inline src/NServiceBus.Persistence.TestRunner/NServiceBus.Persistence.TestRunner.csproj]: Will address

**Comment by bording** [comment]
@danielmarbach The reason you're seeing that happen with the solution file is that SDK-style projects implicitly include Configuration|Platform combos of `Debug|AnyCpu` and `Release|AnyCpu`, so any time you make a change that needs to touch the .sln, it will try to fully populate all of the configurations found in the project.

The thing to keep in mind here is that the values for `Configuration` and `Platform` are mostly just arbitrary strings. What actually matters is what the value of `PlatformTarget` is. 

So for the SDK-style projects, I would suggest letting them have their AnyCpu option in the solution file, and as long you've got the `PlatformTarget` value set correctly, you should be fine.

**Comment by SimonCropp** [comment]
@danielmarbach need any help?

**Comment by danielmarbach** [comment]
Closed in favour of https://github.com/Particular/NServiceBus.Persistence.ServiceFabric/pull/40
